iModelCore/ECPresentation/Tests/NonPublished/Unit/Content/ContentQueryResultsReaderTests.cpp

```diff
@@ -25,6 +25,12 @@ struct ContentQueryResultsReaderTests : QueryExecutorTests
         {
         return ContentDescriptor::Create(*m_connection, *m_ruleset, RulesetVariables(), *NavNodeKeyListContainer::Create(), ContentDisplayType::Undefined, contentFlags, contentFlags);
         }
+    
+    ContentQueryContractPtr CreateContract(uint64_t id, ContentDescriptorPtr descriptor, ECClassCP ecClass, ComplexQueryBuilderPtr queryInfo,
+        bool skipCompositePropertyFields = false, bool skipXToManyRelatedContentFields = false) const
+        {
+        return ContentQueryContract::Create(id, *descriptor, ecClass, *queryInfo, *m_schemaHelper, *m_rulesPreprocessor, nullptr, {}, skipCompositePropertyFields, skipXToManyRelatedContentFields);
+        }

```

Comment:
A function should prefer taking a reference rather than a pointer, if it expects the value to never be NULL. I suggest changing `descriptor` to `ContentDescriptorCR` and `queryInfo` to `ComplexQueryBuilderR`.